In [13]:
import pandas as pd
from scholarly import scholarly
from sklearn.feature_extraction.text import CountVectorizer
from py_stringmatching.similarity_measure.soft_tfidf import SoftTfIdf
import json

In [4]:
pure_researchers = pd.read_excel("../data/pure_researchers.xls")
pure_researchers.head()

1 Name 2 Name > First name 3 Name > Last name  \
0        R. Abdel Kader                  R.        Abdel Kader   
1           Tauseef Ali            Tauseef                 Ali   
2             Robin Aly               Robin                Aly   
3  Juan Amiguet-Vercher                Juan    Amiguet-Vercher   
4        N.L.G. Anciaux              N.L.G.            Anciaux   

         11.1 Organisations > Organisational unit[1]  \
0  Databases (Former) - Former organisational uni...   
1  Datamanagement & Biometrics - Former organisat...   
2  Databases (Former) - Former organisational uni...   
3  Databases (Former) - Former organisational uni...   
4  Databases (Former) - Former organisational uni...   

         11.2 Organisations > Organisational unit[2]  \
0                                                NaN   
1  Services, Cybersecurity & Safety - Former orga...   
2  Human Media Interaction - Former organisationa...   
3                                                NaN   
4                                                NaN   

         11.3 Organisations > Organisational unit[3]  \
0                                                NaN   
1  Services, Cybersecurity & Safety - Former orga...   
2  Databases (Former) - Former organisational uni...   
3                                                NaN   
4                                                NaN   

         11.4 Organisations > Organisational unit[4]  \
0                                                NaN   
1  Services, Cybersecurity & Safety - Former orga...   
2  Databases (Former) - Former organisational uni...   
3                                                NaN   
4                                                NaN   

         11.5 Organisations > Organisational unit[5]  \
0                                                NaN   
1  Signals and Systems (SAS) - Former organisatio...   
2  Human Media Interaction - Former organisationa...   
3                                                NaN   
4                                                NaN   

  11.6 Organisations > Organisational unit[6]  \
0                                         NaN   
1                                         NaN   
2                                         NaN   
3                                         NaN   
4                                         NaN   

  11.7 Organisations > Organisational unit[7]  \
0                                         NaN   
1                                         NaN   
2                                         NaN   
3                                         NaN   
4                                         NaN   

  11.8 Organisations > Organisational unit[8]  \
0                                         NaN   
1                                         NaN   
2                                         NaN   
3                                         NaN   
4                                         NaN   

  11.9 Organisations > Organisational unit[9]  
0                                         NaN  
1                                         NaN  
2                                         NaN  
3                                         NaN  
4                                         NaN

In [5]:
#Explore OUTPut results

In [6]:
pure_output = pd.read_excel("../data/Output_DMB_tbv_studenten_project-20_09_21.xls")
pure_output.head()

ID-0                                     Contributors-1  \
0  90484081  Alessandro Abate, AuthorExternal person, Carlo...   
1   5102579  R. Abdel Kader, AuthorInternal person, Maurice...   
2   5123808  Riham Abdel Kader (R. Abdel Kader), AuthorInte...   
3   5326344  R. Abdel Kader, AuthorInternal person, Maurice...   
4   5413697  Riham Abdel Kader (R. Abdel Kader), AuthorInte...   

  Publication statuses and dates > Date > Year-2  \
0                                           2018   
1                                           2007   
2                                           2011   
3                                           2010   
4                                           2009   

    Title of the contribution in original language-3 Journal > ISSN-4  \
0  Modelling Smart Buildings Using Fault Maintena...              NaN   
1  Overview of query optimization in XML database...              NaN   
2             ROX: Run-Time Optimization of XQueries              NaN   
3        Run-time Optimization for Pipelined Systems              NaN   
4             ROX: Run-Time Optimization of XQueries              NaN   

  Journal > Title-5  \
0               NaN   
1               NaN   
2               NaN   
3               NaN   
4               NaN   

  Electronic version(s) of this work > DOI (Digital Object Identifier)-6  \
0                        10.1007/978-3-030-02227-3_8                       
1                                                NaN                       
2                                                NaN                       
3                                                NaN                       
4                            10.1145/1559845.1559910                       

  Current publication status > Status-7  \
0                             Published   
1                             Published   
2                             Published   
3                             Published   
4                             Published   

  Title of the host publication in original language-8  \
0                   Computer Performance Engineering     
1                                                NaN     
2                                                NaN     
3  Proceedings of the IV Alberto Mendelzon Worksh...     
4                                         SIGMOD '09     

          SEP output type-9  \
0          Conference paper   
1  Professional publication   
2                      Book   
3          Conference paper   
4          Conference paper   

                                            Event-10 Volume-11  \
0  15th European Workshop on Computer Performance...       NaN   
1                                                NaN       NaN   
2                                                NaN       NaN   
3  IV Alberto Mendelzon Workshop on Foundations o...       NaN   
4  35th ACM SIGMOD International Conference on Ma...       NaN   

  Issue number-12 Pages (from-to)-13  
0             NaN            110-125  
1             NaN                NaN  
2             NaN                NaN  
3             NaN         15.1-15.12  
4             NaN            615-626

In [7]:
#Maurice index 113
target_index = 113 #Maurice  
#Google scholar ID JireFzMAAAAJ
target_row = pure_researchers.iloc[target_index,]
print(target_row)

1 Name                                                                        Maurice van Keulen
2 Name > First name                                                                      Maurice
3 Name > Last name                                                                    van Keulen
11.1 Organisations > Organisational unit[1]                          Datamanagement & Biometrics
11.2 Organisations > Organisational unit[2]                            Digital Society Institute
11.3 Organisations > Organisational unit[3]                                       TechMed Centre
11.4 Organisations > Organisational unit[4]    Databases (Former) - Former organisational uni...
11.5 Organisations > Organisational unit[5]    Databases (Former) - Former organisational uni...
11.6 Organisations > Organisational unit[6]    Databases groep - Former organisational unit. ...
11.7 Organisations > Organisational unit[7]    Information Systems (IS) - Former organisation...
11.8 Organisations > Organisat

In [8]:
#Search in output dataset for "1 Name" being in the author column
target_name = target_row['1 Name']
print(target_name)

Maurice van Keulen


In [9]:
from scholarly import scholarly

# Retrieve the author's data, fill-in, and print
search_query = scholarly.search_author(target_name)
author = scholarly.fill(next(search_query))

'geckodriver' executable needs to be in PATH. 
Fetching only the top 20 coauthors


In [10]:
#Fill publications:
filled_publications = []
for p in author['publications']:
    #Retrieve filled p
    filled_p = scholarly.fill(p)
    #Keep only 'bib'
    filled_publications.append({'bib': filled_p['bib']})

In [14]:
print(filled_publications[0])
with open('filled_maurice_pub.json', 'w') as outfile:
    json.dump(filled_publications, outfile)

{'bib': {'title': 'MonetDB/XQuery: a fast XQuery processor powered by a relational engine', 'pub_year': 2006, 'author': 'Peter Boncz and Torsten Grust and Maurice van Keulen and Stefan Manegold and Jan Rittinger and Jens Teubner', 'conference': 'Proceedings of the 2006 ACM SIGMOD international conference on Management of data', 'pages': '479-490', 'publisher': 'ACM', 'abstract': 'Relational XQuery systems try to re-use mature relational data management infrastructures to create fast and scalable XML database technology. This paper describes the main features, key contributions, and lessons learned while implementing such a system. Its architecture consists of (i) a range-based encoding of XML documents into relational tables,(ii) a compilation technique that translates XQuery into a basic relational algebra,(iii) a restricted (order) property-aware peephole relational query optimization strategy, and (iv) a mapping from XML update statements into relational updates. Thus, this system i

In [16]:
print(len(filled_publications))

199


In [39]:
author['filled_publications'] = [p['bib'] for p in filled_publications]
print(author['filled_publications'][0])
print(author['filled_publications'][0].keys())

{'title': 'MonetDB/XQuery: a fast XQuery processor powered by a relational engine', 'pub_year': 2006, 'author': 'Peter Boncz and Torsten Grust and Maurice van Keulen and Stefan Manegold and Jan Rittinger and Jens Teubner', 'conference': 'Proceedings of the 2006 ACM SIGMOD international conference on Management of data', 'pages': '479-490', 'publisher': 'ACM', 'abstract': 'Relational XQuery systems try to re-use mature relational data management infrastructures to create fast and scalable XML database technology. This paper describes the main features, key contributions, and lessons learned while implementing such a system. Its architecture consists of (i) a range-based encoding of XML documents into relational tables,(ii) a compilation technique that translates XQuery into a basic relational algebra,(iii) a restricted (order) property-aware peephole relational query optimization strategy, and (iv) a mapping from XML update statements into relational updates. Thus, this system implement

In [84]:
scholarly.pprint(scholarly.fill(author['publications'][0]))

{'author_pub_id': 'JireFzMAAAAJ:u5HHmVD_uO8C',
 'bib': {'abstract': 'Relational XQuery systems try to re-use mature '
                     'relational data management infrastructures to create '
                     'fast and scalable XML database technology. This paper '
                     'describes the main features, key contributions, and '
                     'lessons learned while implementing such a system. Its '
                     'architecture consists of (i) a range-based encoding of '
                     'XML documents into relational tables,(ii) a compilation '
                     'technique that translates XQuery into a basic relational '
                     'algebra,(iii) a restricted (order) property-aware '
                     'peephole relational query optimization strategy, and '
                     '(iv) a mapping from XML update statements into '
                     'relational updates. Thus, this system implements all '
                     'essential XM

In [99]:
#Check difference between double entries
for p in author['filled_publications']:
    if p['bib']['title'] == "Probabilistic Data Integration" or p['bib']['title'] == "Probabilistic Data Integration.":
        print(p['bib'])

{'title': 'Probabilistic Data Integration.', 'pub_year': 2019, 'author': 'Maurice Van Keulen', 'pages': '1-9', 'abstract': 'Probabilistic data integration (PDI) is a specific kind of data integration where integration problems such as inconsistency and uncertainty are handled by means of a probabilistic data representation. The approach is based on the view that data quality problems (as they occur in an integration process) can be modeled as uncertainty (van Keulen 2012) and this uncertainty is considered an important result of the integration process (Magnani and Montesi 2010). The PDI process contains two phases (see Figure 1):(i) a quick partial integration where certain data quality problems are not solved immediately, but explicitly represented as uncertainty in the resulting integrated data stored in a probabilistic database;(ii) continuous improvement by using the data—a probabilistic database can be queried directly resulting in possible or approximate answers (Dalvi et al 200

In [78]:
print(author.keys())

dict_keys(['container_type', 'filled', 'source', 'scholar_id', 'url_picture', 'name', 'affiliation', 'email_domain', 'interests', 'citedby', 'coauthors', 'publications', 'cites_per_year', 'citedby5y', 'hindex', 'hindex5y', 'i10index', 'i10index5y'])


In [79]:
print(author['publications'])

[{'container_type': 'Publication', 'source': <PublicationSource.AUTHOR_PUBLICATION_ENTRY: 'AUTHOR_PUBLICATION_ENTRY'>, 'bib': {'title': 'MonetDB/XQuery: a fast XQuery processor powered by a relational engine', 'pub_year': '2006'}, 'filled': False, 'author_pub_id': 'JireFzMAAAAJ:u5HHmVD_uO8C', 'num_citations': 404, 'citedby_url': 'https://scholar.google.com/scholar?oi=bibs&hl=en&cites=14607728874914518645,1649838489153578222', 'cites_id': ['14607728874914518645', '1649838489153578222']}, {'container_type': 'Publication', 'source': <PublicationSource.AUTHOR_PUBLICATION_ENTRY: 'AUTHOR_PUBLICATION_ENTRY'>, 'bib': {'title': 'Staircase join: Teach a relational DBMS to watch its (axis) steps', 'pub_year': '2003'}, 'filled': False, 'author_pub_id': 'JireFzMAAAAJ:u-x6o8ySG0sC', 'num_citations': 243, 'citedby_url': 'https://scholar.google.com/scholar?oi=bibs&hl=en&cites=5425607666846947634,4978347757461060471,1670518288144459834', 'cites_id': ['5425607666846947634', '4978347757461060471', '16705

In [40]:
#print(author)


# Print the titles of the author's publications
target_publications_title = [pub['title'] for pub in author['filled_publications']]
target_publications_title.sort()
for f in target_publications_title:
    print(f)
print(len(target_publications_title))

'''
# Take a closer look at the first publication
pub = scholarly.fill(author['publications'][0])
print(pub)

# Which papers cited that publication?
print([citation['bib']['title'] for citation in scholarly.citedby(pub)])
'''

//Rondje Zilverling: COMMIT/TimeTrails
06472 Abstracts Collection--XQuery Implementation Paradigms
06472 Executive Summary--XQuery Implementation Paradigms
08421 Abstracts Collection--Uncertainty Management in Information Systems
08421 Working Group: Imprecision, Diversity and Uncertainty: Disentangling Threads in Uncertainty Management
08421 Working Group: Report of the Probabilistic Databases Benchmarking
A Hybrid Text Classification and Language Generation Model for Automated Summarization of Dutch Breast Cancer Radiology Reports
A Syntax-Directed Editor for TM
A benchmark for online non-blocking schema transformations
A framework for representation, validation and implementation of database application semantics
A generic open world named entity disambiguation approach for tweets
A hybrid approach for robust multilingual toponym extraction and disambiguation
A possible world approach to uncertain relational data
A probabilistic XML approach to data integration
A probabilistic datab

"\n# Take a closer look at the first publication\npub = scholarly.fill(author['publications'][0])\nprint(pub)\n\n# Which papers cited that publication?\nprint([citation['bib']['title'] for citation in scholarly.citedby(pub)])\n"

In [31]:
#Keep Pure Output rows containing target_name in the Contributors-1 column

In [19]:
filter_name = [aut.find(target_name)!=-1 for aut in pure_output["Contributors-1"]]
target_pure_output = pure_output[filter_name]
target_pure_output.head()

ID-0                                     Contributors-1  \
1   5102579  R. Abdel Kader, AuthorInternal person, Maurice...   
3   5326344  R. Abdel Kader, AuthorInternal person, Maurice...   
4   5413697  Riham Abdel Kader (R. Abdel Kader), AuthorInte...   
6   5515633  R. Abdel Kader, AuthorInternal person, Peter B...   
53  5989051  Peter M.G. Apers (Peter Maria Gerardus Apers),...   

   Publication statuses and dates > Date > Year-2  \
1                                            2007   
3                                            2010   
4                                            2009   
6                                            2010   
53                                           1994   

     Title of the contribution in original language-3 Journal > ISSN-4  \
1   Overview of query optimization in XML database...              NaN   
3         Run-time Optimization for Pipelined Systems              NaN   
4              ROX: Run-Time Optimization of XQueries              NaN   
6   ROX: The Robustness of a Run-time XQuery Optim...              NaN   
53  Impress/UT-Soft-D34-001-R1, TMQL Implementatio...              NaN   

   Journal > Title-5  \
1                NaN   
3                NaN   
4                NaN   
6                NaN   
53               NaN   

   Electronic version(s) of this work > DOI (Digital Object Identifier)-6  \
1                                                 NaN                       
3                                                 NaN                       
4                             10.1145/1559845.1559910                       
6                           10.1109/ICDE.2010.5447765                       
53                                                NaN                       

   Current publication status > Status-7  \
1                              Published   
3                              Published   
4                              Published   
6                              Published   
53                             Published   

   Title of the host publication in original language-8  \
1                                                 NaN     
3   Proceedings of the IV Alberto Mendelzon Worksh...     
4                                          SIGMOD '09     
6   Proceedings of the 26th International Conferen...     
53                                                NaN     

           SEP output type-9  \
1   Professional publication   
3           Conference paper   
4           Conference paper   
6           Conference paper   
53     Other research output   

                                             Event-10 Volume-11  \
1                                                 NaN       NaN   
3   IV Alberto Mendelzon Workshop on Foundations o...       NaN   
4   35th ACM SIGMOD International Conference on Ma...       NaN   
6   26th International Conference on Data Engineer...       NaN   
53                                                NaN       NaN   

   Issue number-12 Pages (from-to)-13  
1              NaN                NaN  
3              NaN         15.1-15.12  
4              NaN            615-626  
6              NaN          1185-1188  
53             NaN                NaN

In [33]:
print(target_pure_output.shape)

(188, 14)


# Similarity matching between values in target_pure_output and values in target_publication

We have to find what entry we miss in target_pure_output. Since this has 188 entry and on GScholar there are 199, ideally we have to find 11 missing entries.

## Bag of words

In [41]:
print(target_pure_output.shape)

(188, 14)


In [46]:
#Get list of titles
pure_titles = list(target_pure_output['Title of the contribution in original language-3'].array)
pure_authors = list(target_pure_output['Contributors-1'].array) 
pure_year = list(target_pure_output['Publication statuses and dates > Date > Year-2'].array)
pure_conference = list(target_pure_output['Title of the host publication in original language-8'].array) 

#dict_keys(['title', 'pub_year', 'author', 'conference', 'pages', 'publisher', 'abstract'])
target_pure_output_list = []
for i in range(target_pure_output.shape[0]):
    add_dict = {
        'title':pure_titles[i],
        'pub_year':pure_year[i],
        'author':pure_authors[i],
        'conference':pure_conference[i]
    }

    target_pure_output_list.append(add_dict)

for f in target_pure_output_list:
    print(f)

{'title': 'Overview of query optimization in XML database systems', 'pub_year': '2007', 'author': 'R. Abdel Kader, AuthorInternal person, Maurice van Keulen, AuthorInternal person', 'conference': nan}
{'title': 'Run-time Optimization for Pipelined Systems', 'pub_year': '2010', 'author': 'R. Abdel Kader, AuthorInternal person, Maurice van Keulen, AuthorInternal person, Peter Boncz, AuthorExternal person, Stefan Manegold, AuthorExternal person', 'conference': 'Proceedings of the IV Alberto Mendelzon Workshop on Foundations of Data Management (AMW2010)'}
{'title': 'ROX: Run-Time Optimization of XQueries', 'pub_year': '2009', 'author': 'Riham Abdel Kader (R. Abdel Kader), AuthorInternal person, Peter Boncz (Peter A. Boncz), AuthorExternal person, Stefan Manegold, AuthorExternal person, Maurice van Keulen, AuthorInternal person', 'conference': "SIGMOD '09"}
{'title': 'ROX: The Robustness of a Run-time XQuery Optimizer Against Correlated Data', 'pub_year': '2010', 'author': 'R. Abdel Kader, 

In [35]:
vectoriser = CountVectorizer()
tpo_bag_of_words = vectoriser.fit_transform(target_pure_output_list).toarray()
print(tpo_bag_of_words)

[[0 0 0 ... 0 0 0]
 [0 0 0 ... 0 0 0]
 [0 0 0 ... 0 0 0]
 ...
 [0 0 0 ... 0 0 0]
 [0 0 0 ... 0 0 0]
 [0 0 0 ... 0 0 0]]


In [70]:
# For each word in target_pure_output, compute similarity with each title in target_publications_title:
#Build SoftTfIdf on google scholar results

soft_tfidf = SoftTfIdf()

#Original W: "Overview of query optimization in XML database systems"
#w_compared = target_pure_output_list[0]

#Typo W: "Overview of query optimization in XML database system"
w_compared = "IMPrECISE: Good-is-good-enough data"
#w_compared = "Point of interest to region of interest conversion"

w_compared_list = w_compared.split()
for w in target_publications_title:
    score = soft_tfidf.get_raw_score(w_compared_list, w.split())
    if score > 0.6:
        print(f'"{w_compared}" and "{w}", score: {score}"')

"IMPrECISE: Good-is-good-enough data" and "IMPrECISE: Good-is-good-enough Data Integration", score: 0.6885303623991943"
"IMPrECISE: Good-is-good-enough data" and "IMPrECISE: Good-is-good-enough data integration", score: 0.6546536707079772"


In [56]:
#Loop over google scholar results. If there's no match, add it to the unmatched set
unmatched_titles = set()
treshold = 0.8
soft_tfidf = SoftTfIdf()

for w in target_publications_title:
    #Compute similarity with all word in tarfet_pure_output
    w_list = w.split()
    #Counter of strings with similarity higher than score
    ctr = 0
    for w_c in target_pure_output_list:
        w_c_list = w_c.split()
        if soft_tfidf.get_raw_score(w_c_list, w_list) >= treshold:
            ctr = ctr + 1
            #print(f'Add match {w_c}')
    if ctr == 0:
        #Means there's no match
        #print(f'ctr: {ctr}')
        unmatched_titles.add(w)



{'To use or not to use: guidelines for researchers using data from online social networking sites', 'NLPIT 2016 Chairs’ Welcome', 'Databases and Information Systems in the AI Era: Contributions from ADBIS, TPDL and EDA 2020 Workshops and Doctoral Consortium', 'The IMPRESS DDT: A database design toolbox based on a formal specification language', 'Imprecision, Diversity and Uncertainty: Disentangling Threads in Uncertainty Management: 08421 Working Group', 'Data-Driven Process Discovery and Analysis: 8th IFIP WG 2.6 International Symposium, SIMPDA 2018, Seville, Spain, December 13–14, 2018, and 9th International Symposium, SIMPDA …', 'Workshop FlexDBIST 2011', 'Using element clustering to increase the efficiency of xml schema matching', 'Interactive Explanations of Internal Representations of Neural Network Layers: An Exploratory Study on Outcome Prediction of Comatose Patients', 'Ch. VI. 2 Programmatuur', 'XML: De eend met de gouden eieren?', 'Supporting positional predicates in efficie

In [57]:
print(unmatched_titles)
print(len(unmatched_titles))

{'To use or not to use: guidelines for researchers using data from online social networking sites', 'NLPIT 2016 Chairs’ Welcome', 'Databases and Information Systems in the AI Era: Contributions from ADBIS, TPDL and EDA 2020 Workshops and Doctoral Consortium', 'The IMPRESS DDT: A database design toolbox based on a formal specification language', 'Imprecision, Diversity and Uncertainty: Disentangling Threads in Uncertainty Management: 08421 Working Group', 'Data-Driven Process Discovery and Analysis: 8th IFIP WG 2.6 International Symposium, SIMPDA 2018, Seville, Spain, December 13–14, 2018, and 9th International Symposium, SIMPDA …', 'Workshop FlexDBIST 2011', 'Using element clustering to increase the efficiency of xml schema matching', 'Interactive Explanations of Internal Representations of Neural Network Layers: An Exploratory Study on Outcome Prediction of Comatose Patients', 'Ch. VI. 2 Programmatuur', 'XML: De eend met de gouden eieren?', 'Supporting positional predicates in efficie

# List possible warning 

Idea: Create a file (csv?) with all the warning entries

Entry format: Pure data, GScholar data, warning code

- Same name in Pure (D)
- Entry in pure not matching any in GScholar (M)
- Name missing in pure (NM)
- Same name in pure but different:
    - Author (AM)
    - Conference (CM)
    - Pub year (YM)

In [ ]:
#use author['filled_publication] for Gscholar, keys; ['title', 'pub_year', 'author', 'conference', 'pages', 'publisher', 'abstract']
#target_pure_output_list for Pure, keys: ['title', 'pub_year', 'author', 'conference']

In [23]:
warning_dict = {'D':[], 'M':[], 'NM':[], 'AM':[], 'CM':[], 'YM':[]}
soft_tfidf = SoftTfIdf()

In [70]:
#Get codes D
treshold = 1.0
warning_dict['D'] = []
for i1, pure1 in enumerate(target_pure_output_list):
    w1_list = pure1['title'].split()
    for i2, pure2 in enumerate(target_pure_output_list):
        if i1 != i2:
            w2_list = pure2['title'].split()
            score = soft_tfidf.get_raw_score(w1_list, w2_list)
            if score >= treshold:
                #print(score)
                '''
                Create entry to contain:
                w1, w1_data, w2, w2_data, code D, similarity score
                '''
                add_dict = {
                    'title_1':pure1['title'],
                    'pub_year_1':pure1['pub_year'],
                    'author_1':pure1['author'],
                    'conference_1':pure1['conference'],
                    'title_2':pure2['title'],
                    'pub_year_2':pure2['pub_year'],
                    'author_2':pure2['author'],
                    'conference_2':pure2['conference'],
                    'code':'D',
                    'score':score          
                }

                warning_dict['D'].append(add_dict)

In [71]:
print(warning_dict)
with open('../warnings/code_D.json', 'w') as outfile:
    json.dump(warning_dict['D'], outfile, indent=4)

{'D': [{'title_1': 'Finding You on the Internet: an approach for finding on-line presences of people for fraud risk analysis', 'pub_year_1': '2014', 'author_1': 'Henry Been, AuthorExternal person, Maurice van Keulen, AuthorInternal person', 'conference_1': '16th International Conference on Enterprise Information Systems (ICEIS 2014)', 'title_2': 'Revisiting the formal foundation of Probabilistic Databases', 'pub_year_2': '2015', 'author_2': 'B. Wanders, AuthorInternal person, Maurice van Keulen, AuthorInternal person', 'conference_2': 'Proceedings of the 2015 Conference of the International Fuzzy Systems Association and the European Society for Fuzzy Logic and Technology, IFSA-EUSFLAT 2015', 'code': 'D', 'score': 1.0158562877068968}, {'title_1': 'Finding You on the Internet: an approach for finding on-line presences of people for fraud risk analysis', 'pub_year_1': '2014', 'author_1': 'Henry Been, AuthorExternal person, Maurice van Keulen, AuthorInternal person', 'conference_1': '16th 

In [72]:
#Get codes NM Name missing in Pure
treshold=0.8
warning_dict['NM'] = []
for p in author['filled_publications']:
    #Compute similarity with all word in tarfet_pure_output
    w_list = p['title'].split()
    #Counter of strings with similarity higher than score
    ctr = 0
    for pure_o in target_pure_output_list:
        w_c_list = pure_o['title'].split()
        if soft_tfidf.get_raw_score(w_c_list, w_list) >= treshold:
            ctr = ctr + 1
            #print(f'Add match {w_c}')
    if ctr == 0:
        #Means there's no match
        #print(f'ctr: {ctr}')
        add_dict = {
                    'title_1':p['title'],
                    'pub_year_1':p.get('pub_year', ' '),
                    'author_1':p['author'], 
                    'conference_1':p.get('conference', ' '),
                    'title_2':'',
                    'pub_year_2':'',
                    'author_2':'',
                    'conference_2':'',
                    'code':'M',
                    'score':''           
                }
        warning_dict['NM'].append(add_dict)

In [73]:
print(warning_dict['NM'])
with open('../warnings/code_NM.json', 'w') as outfile:
    json.dump(warning_dict['NM'], outfile, indent=4)

[{'title_1': 'Using element clustering to increase the efficiency of xml schema matching', 'pub_year_1': 2006, 'author_1': 'Marko Smiljanic and Maurice van Keulen and Willem Jonker', 'conference_1': 'Data Engineering Workshops, 2006. Proceedings. 22nd International Conference on', 'title_2': '', 'pub_year_2': '', 'author_2': '', 'conference_2': '', 'code': 'M', 'score': ''}, {'title_1': 'The IMPRESS DDT: A database design toolbox based on a formal specification language', 'pub_year_1': 1994, 'author_1': 'Jan Flokstra and Maurice Van Keulen and Jacek Skowronek', 'conference_1': ' ', 'title_2': '', 'pub_year_2': '', 'author_2': '', 'conference_2': '', 'code': 'M', 'score': ''}, {'title_1': 'Need4tweet: a twitterbot for tweets named entity extraction and disambiguation', 'pub_year_1': 2015, 'author_1': 'Mena Habib and Maurice van Keulen', 'conference_1': ' ', 'title_2': '', 'pub_year_2': '', 'author_2': '', 'conference_2': '', 'code': 'M', 'score': ''}, {'title_1': 'To use or not to use: 

In [74]:
#Get codes M Entry in pure not matching any in GScholar
warning_dict['M'] = []
treshold=0.8
for p in target_pure_output_list:
    #Compute similarity with all word in tarfet_pure_output
    w_list = p['title'].split()
    #Counter of strings with similarity higher than score
    ctr = 0
    for pure_o in author['filled_publications']:
        w_c_list = pure_o['title'].split()
        if soft_tfidf.get_raw_score(w_c_list, w_list) >= treshold:
            ctr = ctr + 1
            #print(f'Add match {w_c}')
    if ctr == 0:
        #Means there's no match
        #print(f'ctr: {ctr}')
        add_dict = {
                    'title_1':p['title'],
                    'pub_year_1':p.get('pub_year', ' '),
                    'author_1':p['author'],
                    'conference_1':p.get('conference', ' '),
                    'title_2':'',
                    'pub_year_2':'',
                    'author_2':'',
                    'conference_2':'',
                    'code':'M',
                    'score':''
                }
        warning_dict['M'].append(add_dict)

In [75]:
print(warning_dict['M'])
with open('../warnings/code_M.json', 'w') as outfile:
    json.dump(warning_dict['M'], outfile, indent=4)

[{'title_1': 'MonetDB/XQuery - Consistent & Efficient Updates on the Pre/Post Plane', 'pub_year_1': '2006', 'author_1': 'Peter Boncz, AuthorExternal person, Jan Flokstra, AuthorInternal person, Torsten Grust, AuthorExternal person, Maurice van Keulen, AuthorInternal person, Stefan Manegold, AuthorExternal person, Sjoerd Mullender, AuthorExternal person, Jan Rittinger, AuthorExternal person, Jens Teubner, AuthorExternal person', 'conference_1': 'Proceedings of the 10th International Conference on Extending Database Technology (EDBT 2006)', 'title_2': '', 'pub_year_2': '', 'author_2': '', 'conference_2': '', 'code': 'M', 'score': ''}, {'title_1': 'Proceedings of the 7th International Symposium on Data-Driven Process Discovery and Analysis (SIMPDA 2017)', 'pub_year_1': '2017', 'author_1': 'Paolo Ceravolo, EditorExternal person, Maurice Van Keulen (Maurice van Keulen), EditorInternal person, Kilian Stoffel, EditorExternal person', 'conference_1': nan, 'title_2': '', 'pub_year_2': '', 'auth